firestore/google/cloud/firestore_v1beta1/_helpers.py

```diff
@@ -150,10 +154,72 @@ def from_string(string):
             if invalid_character in string:
                 raise ValueError('Invalid characters in string.')
         string = string.split('.')
-        return FieldPath(*string)
+        return cls(*string)
+
+    @classmethod
+    def _parse_field_name(cls, api_repr):
+        """Parses the api_repr into the first field name and the rest
+
+        Args:
+            api_repr (str): The unique Firestore api representation.
+
+        Returns:
+            Tuple[str, str]:
+                A tuple with the first field name and the api_repr
+                of the rest.
+        """
+        if api_repr[0] == '.':
+            api_repr = api_repr[1:]
+        if api_repr[0] != '`':  # first field name is simple
+            field_names = api_repr.split('.')
+            return field_names[0], '.'.join(field_names[1:])
+        else:
+            index = 1
+            while index < len(api_repr):
```

Comment:
If given the string
```
a.`
```
won't you return `('a', None)`?

Reply:
I don't think that is a valid api_repr?